wcfsetup/install/files/lib/system/upload/AvatarUploadFileSaveStrategy.class.php

```diff
@@ -78,10 +78,18 @@ public function save(UploadFile $uploadFile)
 
             // shrink avatar if necessary
             try {
+                $newWidth = $newHeight = UserAvatar::AVATAR_SIZE;
+                
+                // Save 4k version if possible.
```

Comment:
Maybe "HiDPI" is a more appropriate name. It's not really related to the term "4K".